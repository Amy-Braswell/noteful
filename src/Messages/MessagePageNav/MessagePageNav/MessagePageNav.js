import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import ApiContext from '../../../ApiContext'
import CircleButton from '../../../CircleButton/CircleButton'
import { findMessage, findMessageMember } from '../../../messages-helpers'
import './MessagePageNav.css'

export default class MessagePageNav extends React.Component {
  static defaultProps = {
    history: {
      goBack: () => { 
        window.history.back()
       }
    },
    match: {
      params: {}
    }
  }
  static contextType = ApiContext

  render() {
    const { messages, members, } = this.context
    const { messageId } = this.props.match.params
    const message = findMessage(messages, messageId) || {}
    const member = findMessageMember(members, message.memberid)
    return (
      <div className='MessagePageNav'>
        <CircleButton
          tag='button'
          role='link'
          onClick={() => this.props.history.goBack()}
          className='MessagePageNav__back-button'
        >
          <FontAwesomeIcon icon={['fa', 'chevron-left']} />
          <br />
          Back
        </CircleButton>
        {member && (
          <h3 className='MessagePageNav__member-name'>
            {member.name}
          </h3>
        )}
      </div>
    )
  }
}
