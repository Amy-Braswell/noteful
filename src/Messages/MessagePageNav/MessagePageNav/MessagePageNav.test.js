import React from 'react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import MessagePageNav from './MessagePageNav'

it('renders without crashing', () => {

  const div = document.createElement('div')
  ReactDOM.render(<MessagePageNav />, div)
  ReactDOM.unmountComponentAtNode(div)
  })

describe(`MessagePageNav component`, () => {
  it('renders a .MessagePageNav by default', () => {
    const wrapper = renderer
    .create(<MessagePageNav />)
    .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it.skip('renders a h3 with member name when in props', () => {
    const props = {
      match: {
        params: {
          messageId: 'test-message-id'
        }
      }
    }
    const context = {
      messages: [{ id: 'test-message-id', memberid: 'test-member-id' }],
      members: [{ id: 'test-member-id', name: 'Important' }]
    }

    const h3 = renderer
    .create(<MessagePageNav {...props} />, context)
    .find('.MessagePageNav__member-name')
    .toJSON()
    expect(h3).toMatchSnapshot()
  })
})