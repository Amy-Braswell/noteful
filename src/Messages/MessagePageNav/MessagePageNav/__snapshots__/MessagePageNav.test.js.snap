// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagePageNav component renders a .MessagePageNav by default 1`] = `
<div
  className="MessagePageNav"
>
  <button
    className="NavCircleButton MessagePageNav__back-button"
    onClick={[Function]}
    role="link"
  >
    <br />
    Back
  </button>
</div>
`;
